Tests/modules/misc/test__weakref.py

```diff
@@ -159,8 +159,15 @@ def %s(self, *args, **kwargs):
                 self.assertEqual(eval('x ' + op + ' 3'), op == '!='); self.assertEqual(called, None)
                 self.assertEqual(eval('3 ' + op + ' x'), op == '!='); self.assertEqual(called, None)
             else:
-                res1, res2 = eval('x ' + op + ' 3'), eval('3 ' + op + ' x')
-                self.assertEqual(called, None)
-                self.assertTrue((res1 == True and res2 == False) or (res1 == False and res2 == True))
+                try:
```

Comment:
Can we use this format instead?
```Python
with self.assertRaises(TypeError):
    eval('x ' + op + ' 3')
```

Also, I assume CPython also throws?

Reply:
Ah; this doesn't throw in CPython, and I think that the whole premise of the test doesn't apply anymore?
```
>>> global called
>>> class C:
...     for method, op in [('__eq__', '=='), ('__gt__', '>'), ('__lt__', '<'), ('__ge__', '>='), ('__le__', '<='), ('__ne__', '!=')]:
...         exec("""
... def %s(self, *args, **kwargs):
...     global called
...     called = '%s'
...     return True
... """ % (method, op))
...
>>> a = C()
>>> import _weakref
>>> x = _weakref.proxy(a)
>>> eval('x < 3')
True
>>> called
'<'
>>> eval('x == 3')
True
>>> called
'=='
```

Reply:
Indeed. It seems like it calls all the underlying operators with Python 3? I guess the test should be inverted.

Reply:
Then let's change the test to look for the new behavior an add an expected failure annotation instead? This looks like a pretty major feature -- I wonder it's under any of the What's new in Python 3.x updates.

Reply:
Agreed, let's update the test to the new behavior. We should file an issue about it and reference it in the test if possible.

I just tried with 3.0 and it looks like the operators "flow through" the proxy. Kind of makes sense since the proxy is supposed to be a stand-in for the actual value. So if it's in any of the release notes it would be in 3.0.

Reply:
I couldn't find it in any release notes, but I believe that it was discussed in this bug: https://bugs.python.org/issue9658 and the one linked at the bottom of it as well. It looks to me like operators in general have been added incrementally to the proxy implementation.